homeassistant/components/onkyo/media_player.py

```diff
@@ -162,116 +198,100 @@ def determine_zones(receiver):
     return out
 
 
-def setup_platform(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: ConfigType,
-    add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
 ) -> None:
-    """Set up the Onkyo platform."""
+    """Set up the Onkyo Platform from config_flow."""
+
     hosts: list[OnkyoDevice] = []
+    platform = entity_platform.async_get_current_platform()
 
-    def service_handle(service: ServiceCall) -> None:
+    def select_output_service(
+        entity: OnkyoDevice, call: ServiceCall
+    ) -> ServiceResponse:
         """Handle for services."""
-        entity_ids = service.data[ATTR_ENTITY_ID]
-        devices = [d for d in hosts if d.entity_id in entity_ids]
-
-        for device in devices:
-            if service.service == SERVICE_SELECT_HDMI_OUTPUT:
-                device.select_output(service.data[ATTR_HDMI_OUTPUT])
+        entity.select_output(call.data[ATTR_HDMI_OUTPUT])
+        return None
 
-    hass.services.register(
-        DOMAIN,
-        SERVICE_SELECT_HDMI_OUTPUT,
-        service_handle,
+    platform.async_register_entity_service(
+        name=SERVICE_SELECT_HDMI_OUTPUT,
         schema=ONKYO_SELECT_OUTPUT_SCHEMA,
+        func=select_output_service,
     )
 
-    if CONF_HOST in config and (host := config[CONF_HOST]) not in KNOWN_HOSTS:
-        try:
-            receiver = eiscp.eISCP(host)
+    host = entry.data[CONF_HOST]
```

Comment:
That's a lot of questions 😅  Let me try to answer:

> Onkyo has 1 host and can have multiple zones

The integration offers support for multiple receivers/hosts, each one has it's own control server and can control multiple zones if the device supports it. It's a one-to-many-to-many relation.

> How does that discovery [to find the other devices] work

It's a command you can send to the onkyo-client software and it returns a list of devices on the network. Newer receivers also support bonjour, but mine does not, so I cannot test that.

> should we start more import flows

In all honesty, I can't really answer that, since I only have 1 device. Maybe I can rig something up to pretend to have more devices configured, but I'd have to check.

Reply:
But I do see that we currently store the host of one of the devices in the entry. Will this device act as the entry point for all other devices via discovery? Or should every device have their own entry?

Reply:
The latter: every device has it's own entry/is it's own host. One device can have multiple zones.